chains/ethereum/server/src/lib.rs

```diff
@@ -238,6 +250,21 @@ impl BlockchainClient for EthereumClient {
         let method_or_position = call_details[1];
         let call_type = call_details[2];
 
+        let block_id = req
+            .block_identifier
+            .as_ref()
+            .map(|block_identifier| -> Result<BlockId> {
+                if let Some(block_hash) = block_identifier.hash.as_ref() {
+                    return BlockId::from_str(block_hash).map_err(|e| anyhow::anyhow!("{e}"));
+                } else if let Some(block_number) = block_identifier.index {
+                    return Ok(BlockId::Number(BlockNumber::Number(U64::from(
+                        block_number,
+                    ))));
+                };
+                bail!("invalid block identifier")
```

Comment:
maybe get the latest block here instead of error.

Reply:
That a good point, this error just happen if you provide an empty block_identifier:
```json
{
    ...
    "block_identifier": {}
}
```
or
```json
{
    ...
    "block_identifier": {
        "index": null,
        "hash": null
    }
}
```
However, in such case, is better provide just null:
```json
{
    ...
    "block_identifier": null
}
```
I don't like to have two valid ways of doing the same thing, because of that I decided to return an error.

Reply:
Also right now if you provide both `index` and `hash`, the `hash` has precedence.